baselines/tamuna/README.md

```diff
@@ -0,0 +1,105 @@
+| `server.p`                 | Describes the probability of server communication while doing local training, in other words, clients will in expectation do 1/p local epochs. Number of local epochs each rounds is synchronized across clients.                                           | 0.333         |
+| `server.uplink_factor`     | Weight of uplink (client to server) communication when calculating communication complexity.                                                                                                                                                                | 1             |
+| `server.downlink_factor`   | Weight of downlink (server to client) communication when calculating communication complexity.                                                                                                                                                              | 1             |
+| `client.learning_rate`     | Learning rate for client local training.                                                                                                                                                                                                                    | 0.01          |
+| `client.eta`               | Learning rate for updating control variates, needs to be between `server.p`/2 and `server.p` * (`server.clients_per_round` * (`server.s` - 1))/(`server.s` * (`server.clients_per_round` - 1)). Usually works very well when simply set to the upper bound. | 0.246         |
+| `meta.n_repeats`           | How many times should the training be repeated from the beginning for both Tamuna and FedAvg. Values bigger than 1 will produce plots that show how the randomness affects the algorithms.                                                                  | 3             |   
+
+## Environment Setup
+
+```bash
+# Create new conda environment with Python 3.9
+conda create -n tamuna python=3.9
+
+# Activate the newly created virtual environment
+conda activate tamuna
+
+# Install poetry using pip
+pip install poetry
+
+# In some cases there are issues when installing the poetry environment if chardet package is not previously installed
+pip install chardet
+
+# Navigate to tamuna directory containing pyproject.toml
+# Install all other packages using poetry
+poetry install
+
+# Note: If you get and error saying permission/access was denied in the middle of package installations, just rerun the poetry install command. It is most probably a bug in poetry. 
+```
+
+
+## Running the Experiments
+
+```bash
+# Default experimental setup in defined in conf/base.yaml, this can be changed if needed  
+poetry run python -m tamuna.main
+
+# Running time for default experimental setup is less than 14min on Intel Core i5-12400F and Nvidia GeForce RTX 3060 Ti.
+```
+
+
+## Expected Results
+
+The resulting directory in `./outputs/` should contain (among other things) `communication_complexity.png` and `loss_accuracy.png`.
```

Comment:
I tested your code and the plots were created perfectly. Could you update the matplotlib code so a title for the figure is included (and maybe adding grid lines too?)

Reply:
I included the titles and added the grid lines.